Context/Adapter/ChainContextAdapter.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Context\Adapter;
+
+use FOS\RestBundle\Context\ContextInterface;
+
+/**
+ * {@inheritdoc}
+ *
+ * @author Ener-Getick <egetick@gmail.com>
+ */
+class ChainContextAdapter implements SerializationContextAdapterInterface, DeserializationContextAdapterInterface, SerializerAwareInterface
+{
+    /**
+     * @var array
+     */
+    private $adapters;
+
+    /**
+     * @var mixed
+     */
+    protected $serializer;
+
+    /**
+     * Constructor.
+     *
+     * @param array $adapters
+     */
+    public function __construct(array $adapters)
```

Comment:
I think we should split them inside the constructor foreach loop .. so that at run time we do not have to do the ``instanceof`` check anymore